src/libstat/backends/redis_backend.c

```diff
@@ -1428,13 +1425,29 @@ rspamd_redis_parse_classifier_opts (struct redis_stat_ctx *backend,
 
 	elt = ucl_object_lookup (obj, "prefix");
 	if (elt == NULL || ucl_object_type (elt) != UCL_STRING) {
+		gchar *redis_object;
+
 		/* Default non-users statistics */
 		if (backend->enable_users || backend->cbref_user != -1) {
-			backend->redis_object = REDIS_DEFAULT_USERS_OBJECT;
+			redis_object = REDIS_DEFAULT_USERS_OBJECT;
 		}
 		else {
-			backend->redis_object = REDIS_DEFAULT_OBJECT;
+			redis_object = REDIS_DEFAULT_OBJECT;
 		}
+
+		/* Prepend classifier name if defined */
```

Comment:
This also might break backward compatibility and finally break all existing stats for users. I don't like it...

Reply:
You are right. 
Then the only way to define separate the redis stat data for different classifier is to properly setup `prefix`.
Personally I would like having a new classifier by unique `name`.
But backward compatibility wins.